src/storage/mvcc/reader/scanner/mod.rs

```diff
@@ -353,12 +355,34 @@ pub fn has_data_in_range<S: Snapshot>(
     right: &Key,
     statistic: &mut CfStatistics,
 ) -> Result<bool> {
-    let iter_opt = IterOptions::new(None, None, true);
+    let iter_opt = IterOptions::new(
+        None,
+        Some(KeyBuilder::from_slice(
+            right.as_encoded(),
+            DATA_PREFIX_KEY.len(),
+            0,
+        )),
+        true,
+    )
+    .set_max_skippable_internal_keys(100);
     let mut iter = snapshot.iter_cf(cf, iter_opt, ScanMode::Forward)?;
-    if iter.seek(left, statistic)? {
-        if iter.key(statistic) < right.as_encoded().as_slice() {
+    match iter.seek(left, statistic) {
+        Ok(valid) => {
+            if valid {
+                if iter.key(statistic) < right.as_encoded().as_slice() {
+                    return Ok(true);
+                }
+            }
+        }
+        Err(e)
+            if e.to_string()
+                .contains("Result incomplete: Too many internal keys skipped") =>
```

Comment:
Should add a test in case of the change of the error message.

Reply:
BTW, it's a little tricky and makes code ugly...

Reply:
see `test_max_skippable_internal_keys_error`

Reply:
I agree. But the rocksdb error type is missing from calling rust-rocksdb API, so I don't any better way to do this.